### ISSUE TITLE
mkbundle cross-compiling issue (Unable to load Assembly)

### ISSUE THREAD

**Issue by carmarri** [mono/mono#7454]
## Steps to Reproduce

1. Take any Mono solution to Linux Mint 32bit (v17), with mono-complete installed
2. Compile it to EXE (successfully)
3. Run EXE (successfully)
4. mkbundle to same platform using
5. mkbundle --simple mydemo.exe -o mydemo (successfully)
6. mkbundle --cross mono-5.10.0-debian-8-x64 mydemo.exe -o mydemo_x86_64 --skip-scan (FAILS!)
`ERROR: Unable to load assembly 'Mono.WebBrowser' referenced by '/home/mc/.mono/targets/mono-5.10.0-debian-8-x64/lib/mono/4.5/System.Windows.Forms.dll'`

The component WebBrowser is not used in the project (just hello-world project with minimal dependencies). I fact, not even that .dll exists in ~/.mono/targets/

If I install Mono 4.8.0 the problem does not occur (same test conditions)

## Current Behavior

Cross compiling to 64 bit platform from a 32bit one fails, reporting a missing file:
```
ERROR: Unable to load assembly `Mono.WebBrowser' referenced by `/home/mc/.mono/targets/mono-5.10.0-debian-8-x64/lib/mono/4.5/System.Windows.Forms.dll'
```

Tested using many different targets with the very same result:
	mono-5.10.0-debian-8-x64
	mono-5.10.0-debian-9-x64
	mono-5.10.0-ubuntu-12.04-x64
	mono-5.10.0-ubuntu-14.04-x64

## Expected Behavior

Should compile properly, as in v4.8.0 does

### On which platforms did you notice this

[ ] macOS
[x ] Linux
[ ] Windows

**Version Used**: Linux Mint 17 Qiana - Cinnamon 32-bit

Mono JIT compiler version 5.10.0.140 (tarball Sat Feb 24 15:37:36 UTC 2018)

### Stacktrace

```
Assembly mydemo.exe loaded successfully.
Attempting to load assembly from: ./I18N.West.dll
Attempting to load assembly from: /home/mc/.mono/targets/mono-5.10.0-debian-8-x64/lib/mono/4.5/I18N.West.dll
Attempting to load assembly from: ./I18N.dll
Attempting to load assembly from: /home/mc/.mono/targets/mono-5.10.0-debian-8-x64/lib/mono/4.5/I18N.dll
Attempting to load assembly from: ./mscorlib
Attempting to load assembly from: /home/mc/.mono/targets/mono-5.10.0-debian-8-x64/lib/mono/4.5/mscorlib
Attempting to load assembly from: ./System.Windows.Forms
Attempting to load assembly from: /home/mc/.mono/targets/mono-5.10.0-debian-8-x64/lib/mono/4.5/System.Windows.Forms
Attempting to load assembly from: ./mscorlib
Attempting to load assembly from: /home/mc/.mono/targets/mono-5.10.0-debian-8-x64/lib/mono/4.5/mscorlib
Attempting to load assembly from: ./System
Attempting to load assembly from: /home/mc/.mono/targets/mono-5.10.0-debian-8-x64/lib/mono/4.5/System
Attempting to load assembly from: ./mscorlib
Attempting to load assembly from: /home/mc/.mono/targets/mono-5.10.0-debian-8-x64/lib/mono/4.5/mscorlib
Attempting to load assembly from: ./Mono.Security
Attempting to load assembly from: /home/mc/.mono/targets/mono-5.10.0-debian-8-x64/lib/mono/4.5/Mono.Security
Attempting to load assembly from: ./mscorlib
Attempting to load assembly from: /home/mc/.mono/targets/mono-5.10.0-debian-8-x64/lib/mono/4.5/mscorlib
Attempting to load assembly from: ./System
Attempting to load assembly from: /home/mc/.mono/targets/mono-5.10.0-debian-8-x64/lib/mono/4.5/System
Attempting to load assembly from: ./System.Configuration
Attempting to load assembly from: /home/mc/.mono/targets/mono-5.10.0-debian-8-x64/lib/mono/4.5/System.Configuration
Attempting to load assembly from: ./mscorlib
Attempting to load assembly from: /home/mc/.mono/targets/mono-5.10.0-debian-8-x64/lib/mono/4.5/mscorlib
Attempting to load assembly from: ./System.Xml
Attempting to load assembly from: /home/mc/.mono/targets/mono-5.10.0-debian-8-x64/lib/mono/4.5/System.Xml
Attempting to load assembly from: ./mscorlib
Attempting to load assembly from: /home/mc/.mono/targets/mono-5.10.0-debian-8-x64/lib/mono/4.5/mscorlib
Attempting to load assembly from: ./System
Attempting to load assembly from: /home/mc/.mono/targets/mono-5.10.0-debian-8-x64/lib/mono/4.5/System
Attempting to load assembly from: ./System.Configuration
Attempting to load assembly from: /home/mc/.mono/targets/mono-5.10.0-debian-8-x64/lib/mono/4.5/System.Configuration
Attempting to load assembly from: ./System
Attempting to load assembly from: /home/mc/.mono/targets/mono-5.10.0-debian-8-x64/lib/mono/4.5/System
Attempting to load assembly from: ./System.Security
Attempting to load assembly from: /home/mc/.mono/targets/mono-5.10.0-debian-8-x64/lib/mono/4.5/System.Security
Attempting to load assembly from: ./mscorlib
Attempting to load assembly from: /home/mc/.mono/targets/mono-5.10.0-debian-8-x64/lib/mono/4.5/mscorlib
Attempting to load assembly from: ./System
Attempting to load assembly from: /home/mc/.mono/targets/mono-5.10.0-debian-8-x64/lib/mono/4.5/System
Attempting to load assembly from: ./Mono.Security
Attempting to load assembly from: /home/mc/.mono/targets/mono-5.10.0-debian-8-x64/lib/mono/4.5/Mono.Security
Attempting to load assembly from: ./System.Xml
Attempting to load assembly from: /home/mc/.mono/targets/mono-5.10.0-debian-8-x64/lib/mono/4.5/System.Xml
Attempting to load assembly from: ./System.Xml
Attempting to load assembly from: /home/mc/.mono/targets/mono-5.10.0-debian-8-x64/lib/mono/4.5/System.Xml
Attempting to load assembly from: ./System.Drawing
Attempting to load assembly from: /home/mc/.mono/targets/mono-5.10.0-debian-8-x64/lib/mono/4.5/System.Drawing
Attempting to load assembly from: ./mscorlib
Attempting to load assembly from: /home/mc/.mono/targets/mono-5.10.0-debian-8-x64/lib/mono/4.5/mscorlib
Attempting to load assembly from: ./System
Attempting to load assembly from: /home/mc/.mono/targets/mono-5.10.0-debian-8-x64/lib/mono/4.5/System
Attempting to load assembly from: ./System.Xml
Attempting to load assembly from: /home/mc/.mono/targets/mono-5.10.0-debian-8-x64/lib/mono/4.5/System.Xml
Attempting to load assembly from: ./System.Runtime.Serialization.Formatters.Soap
Attempting to load assembly from: /home/mc/.mono/targets/mono-5.10.0-debian-8-x64/lib/mono/4.5/System.Runtime.Serialization.Formatters.Soap
Attempting to load assembly from: ./mscorlib
Attempting to load assembly from: /home/mc/.mono/targets/mono-5.10.0-debian-8-x64/lib/mono/4.5/mscorlib
Attempting to load assembly from: ./System.Xml
Attempting to load assembly from: /home/mc/.mono/targets/mono-5.10.0-debian-8-x64/lib/mono/4.5/System.Xml
Attempting to load assembly from: ./Accessibility
Attempting to load assembly from: /home/mc/.mono/targets/mono-5.10.0-debian-8-x64/lib/mono/4.5/Accessibility
Attempting to load assembly from: ./mscorlib
Attempting to load assembly from: /home/mc/.mono/targets/mono-5.10.0-debian-8-x64/lib/mono/4.5/mscorlib
Attempting to load assembly from: ./Mono.WebBrowser
Attempting to load assembly from: /home/mc/.mono/targets/mono-5.10.0-debian-8-x64/lib/mono/4.5/Mono.WebBrowser
Log: 

ERROR: Unable to load assembly `Mono.WebBrowser' referenced by `/home/mc/.mono/targets/mono-5.10.0-debian-8-x64/lib/mono/4.5/System.Windows.Forms.dll'
```

**Comment by carmarri** [comment]
Did anyone ever experienced a similar issue? Any clue on how to fix this?
It is still happening today.

**Comment by tdeeb** [comment]
I'm experiencing this issue on Windows when trying to cross compile to Linux with the **mono-5.10.1-ubuntu-16.04-x64** runtime. I have Mono version 5.10.1.

**Comment by octet8** [comment]
I'm experiencing the same issue on Linux. I have Mono 5.10.1.47 on a 64 bit platform.
```
$ mono --version
Mono JIT compiler version 5.10.1.47 (tarball Tue Apr 17 15:12:35 UTC 2018)
Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
        TLS:           __thread
        SIGSEGV:       altstack
        Notifications: epoll
        Architecture:  amd64
        Disabled:      none
        Misc:          softdebug
        Interpreter:   yes
        LLVM:          supported, not enabled.
        GC:            sgen (concurrent by default)
```

I've tested the following platforms:

```
yocto@linux-laptop:~$ mkbundle --local-targets
Available targets locally:
        default - Current System Mono
        mono-4.8.0-raspbian-8-arm
        mono-5.0.1-raspbian-8-arm
        mono-5.10.1-raspbian-9-arm
        mono-5.12.0-raspbian-9-arm
```

**Comment by nddipiazza** [comment]
Same issue.

**Comment by HinTak** [comment]
Same issue and work around in https://github.com/mono/mono/issues/10490

**Comment by carmarri** [comment]
Does anyone knows what is the status of this? I've tryed to move from 4.8.0 to 5.18.1 and I'm still having dependency issues, as described in #10490.
It looks like this part of the mono project is dead.

**Comment by directhex** [comment]
The missing libraries for WinForms to work will be included in future, and I'm republishing the 6.0.0 cross target to include them too.
